oshmem/mca/spml/spml.h

```diff
@@ -315,24 +315,37 @@ typedef int (*mca_spml_base_module_send_fn_t)(void *buf,
                                               mca_spml_base_put_mode_t mode);
 
 /**
- * The routine transfers the data asynchronously from the source PE to all PEs
- * in the OpenSHMEM job. The routine returns immediately. The source and target
- * buffers are reusable only after the completion of the routine. After the data
- * is transferred to the target buffers, the counter object is updated atomically.
- * The counter object can be read either using atomic operations such as
- * shmem_atomic_fetch or can use point-to-point synchronization routines such as
- * shmem_wait_until and shmem_test.
- *
- *  @param target      The array of remote PE addresses of the objects being written.
- *  @param source      The array of addresses on the local PE holdng the values
- *                     to be written to all other PEs.
+ *  The routine transfers the data asynchronously from the calling PE to all PEs
+ *  in the OpenSHMEM job. The routine returns immediaitely.
+ *
+ *  Upon entry, the following conditions should be true:
+ *  - The counter value is a valid integer value.
+ *  - Target buffers should be ready to receive the data.
+ *
+ *  Upon completion of the routine the following conditions are true on the remote PEs:
+ *  - The counter object is incremented and dest symmetric data object is updated.
+ *  - On the local PE, the counter object and target symmetric data object state
+ *    is undefined.
+ *  - The counter object can be read either using atomic operations such as
+ *    shmem_atomic_fetch or using point-to-point synchronization routines such
+ *    as shmem_wait_until and shmem_test.
+ *
+ *  All invocations of the routine are completed with shmem_quiet call.
+ *  The target and source are reusable only after the completion of the routine.
+ *  No two put_all_nb calls can be active at the same instance. In multithreading case,
+ *  the calls have to be serialized by the User.
```

Comment:
> No two put_all_nb calls can be active at the same instance. In multithreading case, the calls have to be serialized by the User.

How about this ?

` In a multithreaded OpenSHMEM program, the user (the OpenSHMEM program) should ensure the correct ordering of shmemx_alltoall_global calls.`